text/0000-expand-math.md

```diff
@@ -0,0 +1,105 @@
+- Feature Name: Expand `math`
+- Start Date: 2022-02-28
+- RFC PR:
+- Pony Issue:
+
+# Summary
+
+This RFC proposes expanding the standard library's `math` package and the shape it might take to leverage the full extent of Pony. There was prior discussion on [Zulip](https://ponylang.zulipchat.com/#narrow/stream/192795-contribute-to.20Pony/topic/math.20lib) years ago, but I expect some opinions to have changed.
+
+# Motivation
+
+Currently, `math` includes limited functionality. This should be expanded to include math types, constants, and present a structure with further expansion in mind. An expanded `math` library will allow a unified mathematics among Pony developers.
+
+# Detailed design
+
+The primary goals of this initial expansion are: 
+
+1. restructure the `math` package into distinct subpackages; allowing for separation of concerns over continuing to build a monolithic `math` package 
+2. provide common `math` data types; for example, `BigInt`, `Rational`, and `Complex`
+
+## Structure
+
+I propose a structure of distinct subpackages including the following:
+
++ `math/big`: Arbitrary precision numbers
++ `math/series`: Mathematical series
++ `math/constants`: Mathematical constants
++ `math/rational`: `Rational` data type and related functions
++ `math/complex`: `Complex` data type and related functions
++ `math/(x,exp,etc)`: experimental additions, utilities, and effective "catch-all" for matters that do not neatly fit into other subpackages
```

Comment:
why not "just at the top level aka `math`" along with others that are already there?

Reply:
Not sure I understand what you mean, but I suspect you mean dropping the "math" prefix so we have `constants`, `rational`, `complex`, and etc. For that my biggest argument against is that `constants` could be seen as not being "mathematical constants" but rather some way to produce globally immutable "constants" so could be confusing based on that. The alternative for me if we are reducing the subpackaging is that some of these suggestions don't need a package necessary -- a `Constant` primitive can be added to `math` directly without issue.

This gets to the "why" for me including all the piece in one RFC so we can discuss how we want to develop such expansions in the stdlib.

Reply:
I understood Sean's comment in the context of `math/(x,exp,etc)`: instead of introducing a catch-all subpackage of `math`, put those types and functions at the root of the `math` package.

Reply:
I see no issue with `math` itself being the catch-all. If that is the direction we agree is better then I will change the RFC to reflect this decision. I suggested the catch-all subpackage so we are forced to have the conversation about where code that doesn't neatly fit somewhere goes.